seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -143,6 +148,77 @@
             "total_entries": 6
         }
 
+PROJECT_GUID = 'R0001_1kg'
+ADD_DATA_UPDATED_ANVIL_VARIABLES = {
+    "key": "AnVIL_WES",
+    "value": json.dumps({
+        "active_projects": [PROJECT_GUID],
+        "vcf_path": "gs://test_bucket/test_path.vcf",
+        "project_path": "gs://seqr-datasets/v02/GRCh37/AnVIL_WES/{guid}/v1".format(guid=PROJECT_GUID),
+        "projects_to_run": [PROJECT_GUID] })
+}
+ADD_DATA_UPDATE_DAG_TASKS_RESP = {
+            "tasks": [
+                {
+                    "task_id": "create_dataproc_cluster",
+                },
+                {
+                    "task_id": f"pyspark_compute_project_{PROJECT_GUID}",
+                },
+                {
+                    "task_id": "pyspark_compute_variants_AnVIL_WES",
+                },
+                {
+                    "task_id": f"pyspark_export_project_{PROJECT_GUID}",
+                },
+                {
+                    "task_id": "scale_dataproc_cluster",
+                },
+                {
+                    "task_id": f"skip_compute_project_subset_{PROJECT_GUID}",
+                }
+                ],
+            "total_entries": 6
+        }
+
+AIRTABLE_TRACKING_URL = f'{MOCK_AIRTABLE_URL}/appUelDNM3BnWaR7M/AnVIL%20Seqr%20Loading%20Requests%20Tracking'
+
+
+def _setup_responses(responses, update_dag_tasks_resp):
+    # Set up api responses
+    responses.add(responses.POST, AIRTABLE_TRACKING_URL, status=400)
```

Comment:
Maybe these changes are unnecessary because they are testing the identical codes as creating a new Anvil project. What do you think?

Reply:
I don't fully understand the question. You need to test both the create endpoint and the add data endpoint. While they do call the same helper function, we need to test the full behavior for both. I think abstracting this into a helper function is a good way to do that, although I would give it a more descriptive name like `_setup_dag_trigger_responses`

Reply:
also, since ` _test_responses` is a function on the class it might be better to make this a class function as well, just a `@staticmethod`

Reply:
I think `_setup_dag_trigger_responses` can be a class function but `_test_responses` not because it needs to access object data.

Reply:
I was saying that all the functions should be nested in the class, not top-level in the test file. I don't care whether they are staticmethod, classmethod, or regular methods with access to self use whatever is appropriate in each case